### PR TITLE
Pinia

### PR DESCRIPTION
**Pull request by helenahalldiniths**
This will add pinia to the project and a store for handling quizzes

### CODE REVIEW

**Comment by FelixJacobsen** [comment]
Changes made with kevin

1. Added .value on a few places where a ref is used
2. Switched from 'onBeforeMounted' to 'onMounted'
3. Switched current question from being an object to null
4. Removed all data from questions since it's now being fetched.

File: README.md [hunk 1]
```diff
@@ -2,21 +2,27 @@
...
+
 ```sh
 https://github.com/Paefax/paefax-language-app.git
 ```
+
 ### Install dependencies in local repository
+
 ```sh
 npm install
+npm install pinia
```
Comment (LordRekishi) [inline README.md:21]: No needed, enough with npm install

File: src/views/HomeView.vue [hunk 2]
```diff
@@ -11,13 +11,14 @@
 
 <script setup>
 import AnswerItems from "../components/AnswerItems.vue";
-import { ref, onBeforeMount } from "vue";
+import { ref, onMounted } from "vue";
+import { useQuizStore } from "@/stores/quiz";
 const answers = ref([]);
```
Comment (LordRekishi) [inline src/views/HomeView.vue:16]:
```suggestion
import { useQuizStore } from "@/stores/quiz";

const answers = ref([]);
```

File: src/stores/quiz.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { defineStore } from "pinia";
+import { computed, ref } from "vue";
+
+export const useQuizStore = defineStore("quiz", () => {
+  const category = ref("fruit");
+  const questions = ref([]);
+  const idCurrentQuestion = ref(0);
+
+  //Reaktiv
```
Comment (LordRekishi) [inline src/stores/quiz.js:9]:
```suggestion
```

File: src/stores/quiz.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  const numberOfQuestions = computed(() => questions.value.length);
+
+  const setQuestions = (newQuestions) => {
+    console.log("Kevins log3", newQuestions);
```
Comment (LordRekishi) [inline src/stores/quiz.js:13]:
```suggestion
```

File: src/views/HomeView.vue [hunk 5]
```diff
@@ -31,23 +32,52 @@ const checkAnswer = (answer) => {
   }
 };
 
-const nextQuestion = () => {
-  wordToTranslate.value = "Pear";
-  correctAnswer.value = "päron";
+const setQuestionInfo = () => {
+  console.log("Kevins log", currentQuestion.value);
```
Comment (LordRekishi) [inline src/views/HomeView.vue:36]:
```suggestion
```

File: src/views/HomeView.vue [hunk 6]
```diff
@@ -31,23 +32,52 @@ const checkAnswer = (answer) => {
...
+  if (quiz.idCurrentQuestion < quiz.numberOfQuestions - 1) {
+    quiz.nextQuestion();
+    currentQuestion.value = quiz.getCurrentQuestion();
+    setQuestionInfo();
+
+    answeredQuestion.value = false;
+    answeredCorrectly.value = false;
+  } else {
+    // Here we will send the user to the finish/result page.
+    console.log("done");
```
Comment (LordRekishi) [inline src/views/HomeView.vue:62]:
```suggestion
```

File: src/views/HomeView.vue [hunk 7]
```diff
@@ -31,23 +32,52 @@ const checkAnswer = (answer) => {
...
+  }
+};
+
+onMounted(() => {
+  //Set information about quiz (Now hard coded in quiz.js)
+  fetch("questions.json")
+    .then((response) => response.json())
+    .then((data) => {
+      console.log(data);
```
Comment (LordRekishi) [inline src/views/HomeView.vue:71]:
```suggestion
```

File: src/views/HomeView.vue [hunk 8]
```diff
@@ -31,23 +32,52 @@ const checkAnswer = (answer) => {
...
+      quiz.setQuestions(data.questions);
+
+      category.value = quiz.category;
+
+      //Set first question
+      currentQuestion.value = quiz.getCurrentQuestion();
+      console.log("Kevins log2", quiz.getCurrentQuestion());
```
Comment (LordRekishi) [inline src/views/HomeView.vue:78]:
```suggestion
```

**Review by LordRekishi — CHANGES_REQUESTED** [review]
Looks good, let's start with removing some logs and comments. Then I'll look again

**Comment by helenahalldiniths** [comment]
Good job Felix! I looked throught the code and I think the logging really helped me understand it! However I agree with Patrik that they should be removed before merging it into the main branch. I removed some logging and added some comments in order to keep track of what we are doing.

**Pull request merged**